glacier/src/model/index.ts

```diff
@@ -46,7 +46,10 @@ export interface Field {
     readonly table: string;
 }
 
-export type FieldState = Field[];
+export interface FieldState {
+    readonly fields: Field[];
+    readonly dataSource: string;
```

Comment:
This needs to be part of each field, not part of the overall field state, since each field can have a data source, not the model as a whole.